### PR TITLE
Testing/timeproxy

### PR DESCRIPTION
**Pull request by echlebek**
## What is this change?
```
    Use crock to mock out time in schedulerd.
    
    This commit abstracts the time package in schedulerd by using
    timeproxy. In tests, the time is mocked out with crock time.
    
    The result is that these tests no longer need to sleep in order
    to verify that scheduling is working properly.
    
    Test execution has been sped up by a factor of 10-30x as a result.
    
    Signed-off-by: Eric Chlebek <eric@sensu.io>
```

## Does your change need a Changelog entry?

No, this is not a user-facing change.

## Were there any complications while making this change?

This was the first usage of the crock and timeproxy libraries in sensu-go. I had to make some minor changes to those libraries as I worked. Other than that, it was pretty smooth.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

No new documentation is required.

### CODE REVIEW

**Comment by nikkiki** [comment]
```
==================
WARNING: DATA RACE
Write at 0x00c42023a140 by goroutine 14:
  sync/atomic.StoreInt64()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/runtime/race_amd64.s:234 +0xb
  github.com/sensu/sensu-go/vendor/github.com/echlebek/crock.newTimer.func1()
      /home/travis/gopath/src/github.com/sensu/sensu-go/vendor/github.com/echlebek/crock/timer.go:49 +0xb1
  github.com/sensu/sensu-go/vendor/github.com/echlebek/crock.idFunc.Call()
      /home/travis/gopath/src/github.com/sensu/sensu-go/vendor/github.com/echlebek/crock/time.go:123 +0x34
```

```
backend/schedulerd/executor_test.go:40:10: undefined: testSubscriber
backend/schedulerd/executor_test.go:93:10: undefined: testSubscriber
backend/schedulerd/executor_test.go:146:10: undefined: testSubscriber
backend/schedulerd/executor_test.go:203:10: undefined: testSubscriber
```

😱 😱 😱

**Comment by echlebek** [comment]
aw

File: backend/schedulerd/testutil_test.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package schedulerd
+
+type testSubscriber struct {
```
Comment (golangci) [inline backend/schedulerd/testutil_test.go]: type testSubscriber is unused

Comment (echlebek) [reply backend/schedulerd/testutil_test.go]: No it isn't u stupid bot

**Comment by echlebek** [comment]
Looks like there are still some data race issues to resolve in crock.

File: backend/schedulerd/testutil_test.go (hunk 1 above)
Comment (nikkiki) [reply backend/schedulerd/testutil_test.go]: It is both unused and undefined 🤦‍♀️

File: backend/schedulerd/check_scheduler_test.go [hunk 2]
```diff
@@ -1,5 +1,3 @@
-// +build integration,race
```
Comment (nikkiki) [inline backend/schedulerd/check_scheduler_test.go]: It looks like these tests are still integration tests.

**Comment by echlebek** [comment]
This is working now, going to merge it in.

**Pull request merged**